src/k8s-engine/images/imageProcessor.ts

```diff
@@ -504,6 +504,81 @@ export abstract class ImageProcessor extends EventEmitter {
 
     console.log(`Installing kim: kim ${ args.join(' ') }`);
 
+    // Need two loops here. First loop is to verify that we an successfully run `kim builder install`.
+    // If it fails because kubernetes is still running a docker node, we wait and retry, assuming that
+    // eventually it will stop using the docker node and will be running the newer containerd-based node.
+    // You can run `watch kubectl get node -o 'jsonpath={.items[].status.nodeInfo.containerRuntimeVersion}'`
+    // to see how this changes.
+    //
+    // If `kim builder install` fails for any other reason,
+    // this is treated as a fatal but non-terminating error -- the user is advised that they might
+    // need to reset kubernetes in order to build images with nerdctl.
+    //
+    // Once it can successfully run `docker builder install`, it moves to the next loop.
+    // Otherwise it fails on a timeout, again with a logged suggestion to reset kubernetes.
+    while (true) {
+      try {
+        await childProcess.spawnFile(
+          resources.executable('kim'),
+          args,
+          {
+            stdio:       ['ignore', 'pipe', 'pipe'],
+            windowsHide: true,
+          });
+        break;
+      } catch (e) {
+        console.error(`Failed to restart the kim builder: ${ e.message }.`);
+        if (!e.stderr?.includes('Error: container runtime `docker` not supported')) {
+          console.error(`Attempt to run 'kim install builder' => error: '${ e.stderr }'`);
+          console.error('A reset might be necessary to support building images.');
```

Comment:
`console.error` is invisible to the user (they won't look into the logs until something screams in their face); if you intend for this to bubble up, we need to throw a `KubernetesError` or something.

Reply:
No, I didn't think this warranted bubbling up. We can revisit if people report problems.

Reply:
Then the comment on line 514 needs to be changed:

> If `kim builder install` fails for any other reason, this is treated as a fatal but non-terminating error -- the user is advised that they might need to reset kubernetes in order to build images with nerdctl.

Reply:
Reconsidered once I ran into this and reset k8s to make it work...